x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/JoinType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.plan.logical.join;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+
+public enum JoinType implements Writeable {
+    INNER(0, "INNER"),
+    LEFT(1, "LEFT OUTER"),
+    RIGHT(2, "RIGHT OUTER"),
+    FULL(3, "FULL OUTER"),
+    CROSS(4, "CROSS");
```

Comment:
I don't think we need all the join types, especially as we only implement `LEFT OUTER` for now.